packages/x-date-pickers/src/DateField/DateField.tsx

```diff
@@ -154,11 +251,56 @@ DateField.propTypes = {
    * @returns {boolean} If `true` the year will be disabled.
    */
   shouldDisableYear: PropTypes.func,
+  /**
+   * The size of the component.
+   */
+  size: PropTypes.oneOf(['medium', 'small']),
+  style: PropTypes.object,
+  /**
+   * The system prop that allows defining system overrides as well as additional CSS styles.
+   */
+  sx: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool])),
+    PropTypes.func,
+    PropTypes.object,
+  ]),
+  /**
+   * Type of the `input` element. It should be [a valid HTML5 input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).
+   */
+  type: PropTypes.oneOf([
```

Comment:
We can skip this one, we are forcing a value on our side and I don't think people should override it
Or if we want people to override it, we should put it before the spread

Reply:
I believe that we are changing `inputmode` and `type` can easily be changed externally. 🤔 
But I agree, that for this kind of component we should not allow it to change. 😉 👍

Reply:
Nope, we did not look into such solution. But as Flavien mentioned, this change has eventually been reverted in favor of not supporting custom `type` value, because it does not make sense for these components.